utils/firestore.js

```diff
@@ -789,7 +789,9 @@ const filterDB = async (queries, siteCode, isParent) => {
             if(key === 'token') query = query.where('token', '==', queries[key]);
             if(key === 'studyId') query = query.where('state.studyId', '==', queries[key]);
         }
-        const snapshot = await query.where('827220437', operator, siteCode).get();
+        let snapshot = ``
+        queries['allSiteSearch'] === 'true' ? snapshot = await query.get() : snapshot = await query.where('827220437', operator, siteCode).get();
+        const snapshot = await query.where('827220437', operator, siteCodes).get();
```

Comment:
Maybe I am mistaken - Is `snapshot` declared twice here? I see both `let snapshot` and `const snapshot`.
I think we can factor out `snapshot` and `await` in the first part like this (but please test it, I haven't):
`const snapshot = await (queries['allSiteSearch'] === 'true' ? query.get() : query.where('827220437', operator, siteCode).get());`

Reply:
duplicate const snapshot creeped in when i did a git pull & merge. Removed it

Reply:
I like your refactor so Im gonna replace it let with const.